### ISSUE TITLE
Fix `declaration-property-value-no-unknown` false negatives for `var()` usage with typed custom properties

### ISSUE THREAD

**Issue by romainmenke** [stylelint/stylelint#9320]
### What minimal example or steps are needed to reproduce the bug?

```css
@property --brand-color {
  syntax: "<color>";
  inherits: false;
  initial-value: green;
}

@property --brand-image {
  syntax: "<image>";
  inherits: false;
  initial-value: url(#foo);
}

@property --brand-color-or-image {
  syntax: "<color> | <image>";
  inherits: false;
  initial-value: red;
}

.a {
  --brand-color: image-set(linear-gradient(blue, white) 1x, linear-gradient(blue, green) 2x); /* OK (errors as expected) */
  --brand-color: purple; /* OK */
  background-color: var(--brand-color); /* OK */
  background-image: var(--brand-color); /* FAIL (expected an error) */

  --brand-image: image-set(linear-gradient(blue, white) 1x, linear-gradient(blue, green) 2x); /* OK */
  --brand-image: purple; /* OK (errors as expected) */
  background-color: var(--brand-image); /* FAIL (expected an error) */
  background-image: var(--brand-image); /* OK */

  --brand-color-or-image: image-set(linear-gradient(blue, white) 1x, linear-gradient(blue, green) 2x); /* OK */
  --brand-color-or-image: purple; /* OK */
  background-color: var(--brand-color-or-image); /* OK */
  background-image: var(--brand-color-or-image); /* OK */
}
```

### What minimal configuration is needed to reproduce the bug?

```js
export default {
  rules: {
    "declaration-property-value-no-unknown": true
  }
};
```

### How did you run Stylelint?

[Demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcIACAHAThFM0BcCeABALREBGaAhgHZRGQA2EaBwAOlQQQM55U4UAPeAVYgAPAyYA+UQG52nAJZUAFtkU4uwgGYV6XGPI4FlGxXqIA3PQFcYwgOZoYMKkYC+7dqgxZchEnJqWkUAWwoHGBYFbl5+IRFxMIiYGRAjJVV1TR09AwyTKjMLa3o7YRs0egAKAGJtCAgASg8vKh9MbHxiMkoaOghGNCImImTI6OMePkFhUQlB6QIAHwIxcdS5GOU1NA0tAl19Q22inHN6K1t7AmcoVqp2ADoKSc5AvtpJNGENogMcNV6MoYBRhk4KFBFK5AaQyjAADQEADuKg0MCaBAAjAIkcCqKDwZQoTDqnC7EinC4qJiAEwCFoEAD0ACoCAB5ADSBGq2AwaC4BAogpgAiwYBwMCgmJZTJiH2CAyGwhQlRQ9EMzLZXIIspipAoYAA1k4IDZ+t9hNY0NUFRbFmhGayOdy9cYDcbTeaQuFIlawbbeorvk62QAxACCAEkADI80XiyVQIUcPlMGVyx7GO0+lK-X0wf4wQH4wlECEkvhk+FI1HozE4vEgsHl4nQqvkxEEKmuOkM2Ral26zPvIP9DYqtUagfOnW8tD8wXCggJmASqUZ-WGk0Yb1Kpj+m05sYF0MESOx+NitdJlMrhfp4dbz278cFw+BoJvlJnnVu+Vjl8DojMME4mAWRYls2RKQu2sI1iiaKSg2uIEKWLYVnB1YUt2zi9gQ9K-q6I49F+QFDCBJ55gQqpoOqmqzsRz47ma9rKgQ1qfp8+7DKMGxEU+7rbl635+hxAbHt8lH8TO2pMVQ7ggAiIDaIoGoAHIUKEcCIKKWn0U8YBcFwSngBAVCqQ4CAgAmTA4AQsC6DY9D2WwxhoM5MAHG5nCcKIsBgPQYIUOc5lEOgnT+Fc8JEFQEBEOaRpxcijwIAQOAeTAMSeApRimZAFmKA4YZMOEODWVw+AaviOCGeZllPKEABWJnKbAKAmYgPmJJVeDVcoOCiHMIBBZKlWiAiMSiL1-V8AMhUOP8-A0GCUBDYko1eYNIDsDlIDuEAA)

### Which Stylelint-related dependencies are you using?

```json
{
  "stylelint": "latest",
  "stylelint-config-standard": "latest"
}
```

### What did you expect to happen?

I expected typed custom properties in `var()` to error when used in positions where that syntax is invalid.

### What actually happened?

No errors for incorrect `var()` usage where reported

### Do you have a proposal to fix the bug?

`css-tree` doesn't support this validation but we can replace nodes beforehand, like we do for calculations.

We can use the registered syntax to generated values (e.g. `<color>` is replaced with `red`).

There is some complexity with combinators. `<color> | <image>` could be either so we need to generate 2 candidate values and if either passes there shouldn't be an error.

This might be a significant amount of work.

-----

Also see: https://schalkneethling.com/posts/introducing-the-css-property-type-validator-stylelint-plugin/

**Comment by jeddy3** [comment]
> We can use the registered syntax to generated values (e.g. `<color>` is replaced with `red`).
> `<color> | <image>` could be either so we need to generate 2 candidate values and if either passes there shouldn't be an error.

Sounds like a sensible approach to me.

> This might be a significant amount of work.

Let's make the first step a PoC PR to get the ball rolling, so we can gauge the effort required.
